src/main/java/uk/gov/ida/verifyserviceprovider/services/NonMatchingAssertionService.java

```diff
@@ -192,4 +216,25 @@ private String extractLevelOfAssuranceStringFrom(Assertion authnAssertion) {
                 .map(AuthnContextClassRef::getAuthnContextClassRef)
                 .orElseThrow(() -> new SamlResponseValidationException("Expected a level of assurance."));
     }
+
+    private uk.gov.ida.saml.core.domain.AuthnContext getAuthnContextValueFromUri(String authnContextUriValue) {
```

Comment:
I think it'd be worth moving this to a helper class of its own.  Also, it should be made static.

Reply:
No longer required, I have changed the implementation